File: Session8/Java/InsertInLinkedList.java
class InsertInLinkedList
{
	Node head; 
	class Node
	{
		int data;
		Node next;
		Node(int d) 
		{
			data = d; 
			next = null; 
		}
	}

	public void insertionAtHead(int data)
	{
		Node new_node = new Node(data);
		new_node.next = head;
		head = new_node;
	}

	public void insertionAtTail(int new_data)
	{
		if (head == null)
		{
			head = new Node(new_data);
			return;
		}

		Node temp = head;
		while (temp.next != null)
		{
			temp = temp.next;
		}
		temp.next = new Node(new_data);
	}

	public void printList()
	{
		Node temp = head;
		while (temp!=null)
		{
			System.out.print(temp.data+" "+"-->");
			temp = temp.next;
		}
	}

	public static void main(String[] args)
	{
		InsertInLinkedList llist = new InsertInLinkedList();

		llist.insertionAtHead(1);
		llist.insertionAtHead(2);
		llist.insertionAtHead(3);
		llist.insertionAtHead(4);
		llist.insertionAtHead(5);
		
		llist.insertionAtTail(8);
		llist.insertionAtTail(9);

		llist.printList();
	}
}

File: Session8/C++/InsertInLinkedList.cpp
#include <iostream>
using namespace std;

class Node{
public:
    int data;
    Node *next;

    //constructor
    Node(int d)
    {
        data = d;
        next = NULL;
    }
};

void insertionAtHead(Node*&head, int data)
{
    Node *new_Node = new Node(data);
    new_Node -> next = head;
    head = new_Node;
}

void insertionAtTail(Node*&head, int data)
{
    if(head==NULL)
    {
        head = new Node(data);
        return;
    }

    Node*temp = head;
    while(temp->next!=NULL)
    {
        temp = temp->next;
    }
    
    temp->next = new Node(data);
}

void print(Node*head)
{
    Node *temp = head;
    while(temp!=NULL)
    {
        cout<<temp->data<<"-->";
        temp = temp->next;
    }
    cout<<endl;
}

int main() {

    Node* head = NULL;  

    insertionAtHead(head, 1);
    insertionAtHead(head, 2);
    insertionAtHead(head, 3);
    insertionAtHead(head, 4);
    insertionAtHead(head, 5);

    insertionAtTail(head, 8);

    print(head);
    return 0;
}

